app/helpers/uploads_helper.rb

```diff
@@ -24,6 +24,21 @@ def uploads_destroy_params(form1, form2 = nil)
     params
   end
 
+  def uploads_retry_params(form1, form2 = nil)
+    params = {}
+    params["#{form1.object_name}[id]"] = form1.object.id
+    params["#{form2.object_name}[id]"] = form2.object.id if form2
+
+    # retry only the right-most form object
```

Comment:
what is the case that there are 2 forms, and what constitutes the right-most form?

Reply:
So this was a gotcha that took me ~way too long~ a normal amount of time to track down.

So I'm trying to generate form params nested 2 levels in.  Inside a `fields_for :default_feed` _AND_ inside that a `fields_for :itunes_image`.

If I only use the last (itunes_image) form to generate the params, I'd get:

```
podcast[default_feed_attributes][itunes_images_attributes][0][_retry]: 1
podcast[default_feed_attributes][itunes_images_attributes][0][id]: 456
```

And Rails would fail to pass that param, with something like "can't find itunes image with id 456".  The trick was: Rails requires _both_ the feed's ID, and the itunes_image ID to do its magic `accepts_nested_attributes_for`.  So using both forms things work!

```
podcast[default_feed_attributes][id]: 123
podcast[default_feed_attributes][itunes_images_attributes][0][_retry]: 1
podcast[default_feed_attributes][itunes_images_attributes][0][id]: 456
```